aspnetcore/web-api/action-return-types.md

```diff
@@ -10,6 +10,206 @@ uid: web-api/action-return-types
+
+The <xref:Microsoft.AspNetCore.Mvc.IActionResult> return type is appropriate when multiple `ActionResult` return types are possible in an action. The `ActionResult` types represent various HTTP status codes. Any non-abstract class deriving from `ActionResult` qualifies as a valid return type. Some common return types in this category are <xref:Microsoft.AspNetCore.Mvc.BadRequestResult> (400), <xref:Microsoft.AspNetCore.Mvc.NotFoundResult> (404), and <xref:Microsoft.AspNetCore.Mvc.OkObjectResult> (200). Alternatively, convenience methods in the <xref:Microsoft.AspNetCore.Mvc.ControllerBase> class can be used to return `ActionResult` types from an action. For example, `return BadRequest();` is a shorthand form of `return new BadRequestResult();`.
+
+Because there are multiple return types and paths in this type of action, liberal use of the [`[ProducesResponseType]`](xref:Microsoft.AspNetCore.Mvc.ProducesResponseTypeAttribute) attribute is necessary. This attribute produces more descriptive response details for web API help pages generated by tools like [Swagger](xref:tutorials/web-api-help-pages-using-swagger). `[ProducesResponseType]` indicates the known types and HTTP status codes to be returned by the action.
+
+### Synchronous action
+
+Consider the following synchronous action in which there are two possible return types:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultProductsController.cs" id="snippet_GetByIdIActionResult" highlight="7":::
+
+In the preceding action:
+
+* A 404 status code is returned when the product represented by `id` doesn't exist in the underlying data store. The <xref:Microsoft.AspNetCore.Mvc.ControllerBase.NotFound%2A> convenience method is invoked as shorthand for `return new NotFoundResult();`.
+* A 200 status code is returned with the `Product` object when the product does exist. The <xref:Microsoft.AspNetCore.Mvc.ControllerBase.Ok%2A> convenience method is invoked as shorthand for `return new OkObjectResult(product);`.
+
+### Asynchronous action
+
+Consider the following asynchronous action in which there are two possible return types:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/mvc/action-return-types/7.x/WebApiSample/Controllers/ActionResultProductsController.cs" id="snippet_CreateAsyncIActionResult" highlight="9,15":::
+
+In the preceding action:
+
+* A 400 status code is returned when the product description contains "XYZ Widget". The <xref:Microsoft.AspNetCore.Mvc.ControllerBase.BadRequest%2A> convenience method is invoked as shorthand for `return new BadRequestResult();`.
+* A 201 status code is generated by the <xref:Microsoft.AspNetCore.Mvc.ControllerBase.CreatedAtAction%2A> convenience method when a product is created. The following code is an alternative to calling `CreatedAtAction`:
+
+  ```csharp
+  return new CreatedAtActionResult(nameof(GetById), "Products", new { id = product.Id }, product);
```

Comment:
```suggestion
  return new CreatedAtActionResult(nameof(GetById), "Products",
                                                                    new { id = product.Id }, product);
```
Get rid of horizontal scroll bar
<img width="514" alt="image" src="https://user-images.githubusercontent.com/3605364/193922980-a43446fe-940c-4f52-9385-e8b6bc913350.png">